python-sdk/src/astro/files/locations/ftp.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from urllib.parse import urlparse
+
+import smart_open
+from airflow.providers.ftp.hooks.ftp import FTPHook
+
+from astro.constants import FileLocation
+from astro.files.locations.base import BaseFileLocation
+
+
+class FTPLocation(BaseFileLocation):
+    """Handler FTP object store operations"""
+
+    location_type = FileLocation.FTP
+    supported_conn_type = {FTPHook.conn_type}
+
+    @property
+    def hook(self) -> FTPHook:
+        return FTPHook(ftp_conn_id=self.conn_id) if self.conn_id else FTPHook()
+
+    @property
+    def transport_params(self) -> dict:
+        """get FTP credentials for remote file system"""
+        return {}
+
+    @property
+    def paths(self) -> list[str]:
+        """Resolve FTP file paths with prefix"""
+        url = urlparse(self.path)
+        conn = self.hook.get_connection(self.conn_id)
+        uri = conn.get_uri()
+        client = self.hook.get_conn()
+        files = client.nlst("/" + url.netloc + url.path)
+        if len(files) > 0:
+            paths = [uri + file for file in files]
+        else:
```

Comment:
When would this else scenario occur and what does that mean?

Reply:
i am checking whether its file or folder here

Reply:
I think it would be good to add a comment because `len(files) >0` doesn't clarify that

Reply:
will add it